docs/filters.rst

```diff
@@ -0,0 +1,87 @@
+.. _filters:
+
+Filtering
+=========
+
+REST resource filtering is provided via method ``_filter_queryset``. If you want implement your own filter technique
+you can add here your own code. Input params of the method are only ``qs`` which represents data that you want to filter
+(for model resources it is django QuerySet object).
+
+Better way is used pre-implemented managers defined in model ``BaseModelResource``::
+
+    class BaseModelResource(DefaultRESTModelResource, BaseObjectResource):
+        filter_manager = MultipleFilterManager()
+
+        def _filter_queryset(self, qs):
+            if self.filter_manager:
+                return self.filter_manager.filter(self, qs, self.request)
+            else:
+                return qs
+
+There method ``_filter_queryset`` uses manager to filter output data. You can change manager with filter_manager
+attribute. If you set filter_manager to ``None``. Filtering will be disabled. Default filter manager is
+``MultipleFilterManager``.
+
+Filter manager
+--------------
+
+The main purpose of filter manager is parse input data, construct django models Q objects and use it to filter data
+from model QuerySet. Manager has only one public method ``filter(self, resource, qs, request)`` that accepts
+resource object, QuerySet and Django request object.
+
+Manager too must be able to validate filter input, parse this input to identifiers, operators and values and search
+concrete filter classes from filter identifiers. It is responsible for checking if concrete filter is allowed
+inside resource for filtering. For this purpose resource provides ``filter_fields_rfs``. By default it is all fields
+that is customer allowed to read. These fields can be rewritten on model meta or resource with attributes
+``filter_fields`` or ``extra_filters_fields``. Attribute ``filter_fields`` defines list of fields that is allowed
+to filter and ``extra_filters_fields`` extends these fields. There is used same rule as for other rest fields, resource
+attributes rewrite model meta attributes.
```

Comment:
A manager must be able to

1. validate filter specification in GET arguments
2. parse this input into identifiers, operators and values
3. search corresponding filter classes accorring to filter identifiers.
4. check if the specific filter is allowed on the resource, i.e. if it is in ``filter_fields_rfs``

By default, ``filter_fields_rfs`` contains all fields the user is allowed to read. 
These fields can be rewritten in RESTMeta class or on the resource using attributes ``filter_fields`` or ``extra_filters_fields``. 
Attribute ``filter_fields`` defines a list of fields that are allowed to filter by and ``extra_filters_fields`` extends the list.
The rule of resource fields overriding the values from RESTMeta class of the model applies just like with other REST fields attributes.